src/test/java/com/artipie/http/TrafficMetricSliceTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.http;
+
+import com.artipie.http.rq.RequestLine;
+import com.artipie.http.rq.RqMethod;
+import com.artipie.http.rs.RsWithBody;
+import com.artipie.metrics.Metrics;
+import com.artipie.metrics.memory.InMemoryMetrics;
+import com.artipie.metrics.publish.MetricsOutput;
+import hu.akarnokd.rxjava2.interop.CompletableInterop;
+import io.reactivex.Flowable;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test case for {@link TrafficMetricSlice}.
+ *
+ * @since 0.10
+ */
+final class TrafficMetricSliceTest {
+
+    /**
+     * Request metrics key.
+     */
+    private static final String KEY_REQ = "request.body.size";
+
+    /**
+     * Response metrics key.
+     */
+    private static final String KEY_RSP = "response.body.size";
+
+    @Test
+    void updatesMetrics() throws Exception {
+        final Metrics metrics = new InMemoryMetrics();
+        final int size = 100;
+        new TrafficMetricSlice((line, head, body) -> new RsWithBody(body), metrics)
+            .response(
+                new RequestLine(RqMethod.GET, "/foo").toString(),
+                Collections.emptyList(),
+                Flowable.just(ByteBuffer.wrap(new byte[size]))
```

Comment:
@g4s8 I'd suggest to have different size for request and response, otherwise we may mix up response and request in metrics and never know